### ISSUE TITLE
Allow using quoted PEP 677 Callable Type Syntax in  python versions older than 3.11

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#3009]
**Is your feature request related to a problem? Please describe.**
I would like to use the "PEP 677 Callable Type Syntax" within quotes in my type hints, even though I'm using an older version of python. I see that the Union syntax from PEP 604 is allowed within quotes in versions older than 3.10, so I was hoping the same would be true for the new callable syntax .

**Comment by erictraut** [comment]
This should already work. Do you have an example of it not working somewhere?

**Comment by Tomaz-Vieira** [comment]
Apparently it's only happening in second order functions:

```python3
#pyright: strict

def foo(a: "(int) -> float"): # no issues
    raise Exception()

def bar(a: "(int) -> ((str) -> bool)"): #error
    raise Exception()

#                     ^^^^^^^^^^^^^
#6:23 - error: Callable syntax requires Python 3.11 or newer
```

But it works fine if I run pyright with `--pythonversion 3.11`

(version 1.1.219)

**Comment by erictraut** [comment]
Ah yes, that's a bug. It will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.220, which I just published. It will also be included in the next release of pylance.
